### ISSUE TITLE
[BUG] Quoted message in community displayed as "Original message not found" for the original message sender

### ISSUE THREAD

**Issue by fracting** [oxen-io/session-ios#768]
### Code of conduct

- [X] I have read and agree to adhere to the [Code of Conduct](https://github.com/oxen-io/session-ios/blob/master/CODE_OF_CONDUCT.md)

### Self-training on how to write a bug report

- [X] I have learned [how to write a bug report](https://www.browserstack.com/guide/how-to-write-a-bug-report)

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When a message sent from iOS is quoted in a reply, the original sender sees "Original message not found".
Other people in the same community can see the original message being quoted

### Expected Behavior

The original sender should be able to see the original message in the quote.

### Steps To Reproduce

1. Create two fresh new accounts on iOS called Alice and Bob
2. (Whether Alice and Bob are friends seems to have no impact on this bug)
3. Both Alice and Bob join a community with blinded ID enabled, for example, the official Session Community.
4. Alice sends a message "This is Alice, testing session bug"
5. Bob quotes and replies to Alice's message, "Hi Alice, I'm quoting your message from iOS"

As the Screenshot shows, in Alice's view, the original message was incorrectly displayed as "Original message not found", while in Bob's view or any other third party's view, the original message was correctly shown in the quote.

Vise vasa, switching Alice and Bob can reproduce the bug in a symmetric way.

### iOS Version

iOS 16.0

### Session Version

2.2.4

### Anything else?

<img width="823" alt="Screen Shot 2023-01-13 at 6 36 25 pm" src="https://user-images.githubusercontent.com/194342/212266929-9dd94f01-3a94-4a97-b591-beeec96be7ed.png">

**Comment by fracting** [comment]
I have a theory, maybe Alice's local message database stores her own message using her unblinded ID but try to lookup the message using her blinded ID when being quoted.

I got this theory by debugging on the Session Desktop app, but I haven't verified with the iOS app yet.

**Comment by fracting** [comment]
TODO: someone can help to verify if the same bug exists on Android and Desktop.

**Comment by fracting** [comment]
```
quote: Optional(
  SessionMessagingKit.Quote(
    interactionId: 104,
    authorId: "15344781c3da0427cc16ff5178f5521abf2d8b905a8435d756a529969b33380f97",
    timestampMs: 1673595324642,
    body: nil,
    attachmentId: nil
  )
),

quoteAttachment: nil,

linkPreview: nil,

linkPreviewAttachment: nil,

currentUserPublicKey: "05e54602b302fd44ad85bc297f39c93f9e8ac359d621bc0beed3905a96c833dd28",

attachments: nil,

reactionInfo: nil,

cellType: SessionMessagingKit.MessageViewModel.CellType.textOnlyMessage,

authorName: "Alice (152c...f882)",

senderName: Optional("Alice (152c...f882)"),

shouldShowProfile: true,

shouldShowDateHeader: false,

containsOnlyEmoji: Optional(false),

glyphCount: Optional(41),

previousVariant: Optional(SessionMessagingKit.Interaction.Variant.standardOutgoing),

positionInCluster: SessionUIKit.Position.middle,

isOnlyMessageInCluster: true,

isLast: false,

currentUserBlindedPublicKey: nil
```

`currentUserBlindedPublicKey: nil` looks suspicious to me.

**Comment by GittyUpMyHorsey** [comment]
I also have this issue as well on the latest ios and latest session versions.

**Comment by RyanRory** [comment]
Hi @fracting thanks for the feedback and helping us debug this issue, I'll make a quick fix for this.

**Comment by fracting** [comment]
Confirming fixed in Session iOS 2.2.8 for me. Thanks for the fix!

https://github.com/oxen-io/session-desktop/issues/2655 is still reproducible with Session iOS 2.2.8.
